app/src/main/java/com/example/plantsapp/presentation/ui/plantdetail/PlantDetailFragment.kt

```diff
@@ -27,8 +27,18 @@ class PlantDetailFragment : Fragment(R.layout.fragment_plant_detail) {
         super.onViewCreated(view, savedInstanceState)
 
         with(detailViewModel) {
-            plant.observe(viewLifecycleOwner) {
-                showPlantDetail(it)
+            plant.observe(viewLifecycleOwner) { plant ->
+                showPlantDetail(plant)
+
+                binding.btnDelete.setOnClickListener {
+                    detailViewModel.onDelete(plant)
+                }
```

Comment:
How about move this btnDelete.setOnclickListener out of observing data, and remove plant from onDelete params? 
So when you press delete , data will be taken from viewmodel

Reply:
Also you can disable button initially, and enable it when data arrives